src/observable.cpp

```diff
@@ -22,35 +46,58 @@ std::istream& operator>>(std::istream& input, wxString& ws) {
 }
 
 SingletonVar* SingletonVar::getInstance(const std::string& key) {
-  static std::map<std::string, SingletonVar*> instances;
+  static std::unordered_map<std::string, SingletonVar*> instances;
 
   if (instances.find(key) == instances.end()) {
     instances[key] = new SingletonVar();
   }
   return instances[key];
 }
 
+
 /* ObservedVar implementation. */
 
-void ObservedVar::listen(wxWindow* listener, wxEventType ev_type) {
+void ObservedVar::listen(wxEvtHandler* listener, wxEventType ev_type) {
+  const auto& listeners = singleton->listeners;
+  if (listeners.find(listener) != listeners.end())
+    wxLogWarning("Duplicate window listener %",  listener);
   singleton->listeners[listener] = ev_type;
 }
 
-bool ObservedVar::unlisten(wxWindow* listener) {
+bool ObservedVar::unlisten(wxEvtHandler* listener) {
   auto& listeners = singleton->listeners;
   if (listeners.find(listener) == listeners.end()) return false;
   listeners.erase(listener);
   return true;
 }
 
-const void ObservedVar::notify() {
+const void ObservedVar::notify(const std::string& s, void* client_data) {
   auto& listeners = singleton->listeners;
   for (auto l = listeners.begin(); l != listeners.end(); l++) {
```

Comment:
Looking at this more closely, `listeners` doesn't really need to be a map at all. It could just be a `std::vector<std::pair<wxEventHandler*, wxEventType>>`.  Scanning it to remove shouldn't really be a big deal.

Reply:
Indeed, good catch! fixed in last version